src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -652,13 +670,15 @@ protected int GetHashCodeHelper()
                    Hash.Combine(this.ReportSuppressedDiagnostics,
                    Hash.Combine(this.ScriptClassName != null ? StringComparer.Ordinal.GetHashCode(this.ScriptClassName) : 0,
                    Hash.Combine(Hash.CombineValues(this.SpecificDiagnosticOptions),
+                   Hash.Combine(Hash.CombineValues(PerTreeDiagnosticOptions.Keys),
```

Comment:
>PerTreeDiagnosticOptions.Keys [](start = 51, length = 29)

Order of elements in these collections is probably non-deterministic. I wouldn't expect result of this function be deterministic, but I think it must be deterministic for the purposes of GetHashCode.